mitiq/pt/pt.py

```diff
@@ -0,0 +1,163 @@
+            ``None``, the ``executor`` must return an expectation value
+            (float). Otherwise, the ``QuantumResult`` returned by ``executor``
+            is used to compute the expectation of the observable.
+        num_circuits: Number of circuits to be randomly twirled, and averaged.
+
+    Returns:
+        The expectation value estimated with Pauli twirling.
+    """
+    executor = (
+        executor if isinstance(executor, Executor) else Executor(executor)
+    )
+    twirled_circuits = twirl_CNOT_gates(circuit, num_circuits)
+    expvals = executor.evaluate(twirled_circuits, observable)
+    return cast(float, np.average(expvals))
+
+
+def twirl_CNOT_gates(circuit: QPROGRAM, num_circuits: int) -> List[QPROGRAM]:
+    """Generate a list of circuits using Pauli twirling on CNOT gates.
+
+    Args:
+        circuit: The circuit to generate twirled versions of
+        num_circuits: The number of sampled circuits to return
+    """
+    return [_twirl_CNOT_qprogram(circuit) for _ in range(num_circuits)]
+
+
+@noise_scaling_converter
+def _twirl_CNOT_qprogram(circuit: cirq.Circuit) -> cirq.Circuit:
+    return circuit.map_operations(_twirl_single_CNOT_gate)
+
+
+def twirl_CZ_gates(circuit: QPROGRAM, num_circuits: int) -> List[QPROGRAM]:
+    """Generate a list of circuits using Pauli twirling on CZ gates.
+
+    Args:
+        circuit: The circuit to generate twirled versions of
+        num_circuits: The number of sampled circuits to return
+    """
+    return [_twirl_CZ_qprogram(circuit) for _ in range(num_circuits)]
+
+
+@noise_scaling_converter
+def _twirl_CZ_qprogram(circuit: cirq.Circuit) -> cirq.Circuit:
+    return circuit.map_operations(_twirl_single_CZ_gate)
+
+
+def _twirl_single_CNOT_gate(op: cirq.Operation) -> cirq.OP_TREE:
+    """Function which converts a CNOT to a logical equivalent. That is, it
+    converts CNOT operations into the following, leaving all other cirq
+    operations alone.
+
+        --P---⏺---R--
+              |
+        --Q---⊕---S--
+
+    Where P, Q, R, and S are Pauli operators sampled so as to not effect
+    change on the underlying unitary.
+    """
+    if op.gate != cirq.CNOT:
+        return op
+
+    P, Q, R, S = random.choice(CNOT_twirling_gates)
+    control_qubit, target_qubit = op.qubits
+    return [
+        P.on(control_qubit),
+        Q.on(target_qubit),
+        op,
+        R.on(control_qubit),
+        S.on(target_qubit),
+    ]
+
+
+def _twirl_single_CZ_gate(op: cirq.Operation) -> cirq.OP_TREE:
+    """Function which converts a CNOT to a logical equivalent. That is, it
+    converts CNOT operations into the following, leaving all other cirq
+    operations alone.
+
+        --P---⏺---R--
+              |
+        --Q---⏺---S--
+
+    Where P, Q, R, and S are Pauli operators sampled so as to not effect
+    change on the underlying unitary.
+    """
+    if op.gate != cirq.CZ:
+        return op
+
+    P, Q, R, S = random.choice(CZ_twirling_gates)
```

Comment:
Any idea why Codecov complains about `_twirl_single_CZ_gate` but not `_twirl_single_CZ_gate`? 

IMO we don't have to resolve it on this PR, it just raised my curiousity.

Reply:
At first glance, it looks like there's only 1 test for `twirl_CZ_gates` and 3 additional tests for `twirl_CNOT_gates`. 

Maybe Codecov's issue is related to those additional tests? Haven't looked at it in more detail though.